docs/javascript/v2/foundation/security-and-tokens.mdx

```diff
@@ -9,22 +9,96 @@ nav: 1.3
+
+-   You can get the `app_access_key` and `app_secret` from the [developer section](https://dashboard.100ms.live/developer) in your 100ms dashboard.
 -   **room_id**: This is the unique identifier for your room. You can get it from the [rooms page](https://dashboard.100ms.live/rooms) in your dashboard or in the response payload of the [create room server-side API](/server-side/v2/Rooms/create-via-api).
--   **user_id**: This identifier can be used to map a 100ms peer to your own internal user object for business logic. Specify your internal user identifier as the peer’s user_id. If not available, use any random string.
+-   **user_id**: This identifier can be used to map a 100ms peer to your own internal user object for business logic. Specify your internal user identifier as the peer's user_id. If not available, use any random string.
+
+### Use 100ms token endpoint for building integration
+
+You can use the token endpoint from your 100ms dashboard while building integration with 100ms. This acts as a tool enabling front-end developers to complete the integration without depending on the backend developers to set up a token generation backend service.
+
+**URL format:** `https://prod-in2.100ms.live/hmsapi/<your-100ms-subdomain>.app.100ms.live/api/token`
+
+100ms token endpoint can generate an app token with the inputs passed, such as room_id, role, & user_id (optional - your internal user identifier as the peer's user_id). You can use [jwt.io](https://jwt.io/) to validate whether the app token contains the same input values.
+
+<PostRequest title="https://prod-in2.100ms.live/hmsapi/johndoe.app.100ms.live/api/token" />
+<Request id="req-comp-0">
+
+```bash
+curl --location --request POST 'https://prod-in2.100ms.live/hmsapi/johndoe.app.100ms.live/api/token' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "room_id":"633fcdd84208780bf665346a",
+    "role":"host",
+    "user_id":"1234"
+}'
+```
+
+</Request>
+<ResponseBox id="resp-0" status="200 OK">
+
+```json
+{
+    "token": "***************************",
+    "msg": "token generated successfully",
+    "status": 200,
+    "success": true,
+    "api_version": "2.0.192"
+}
+```
+
+</ResponseBox>
 
-### Code sample: Generate app token
+You can directly add this to your client-side implementation, for example:
+
+```jsx
+const response = await fetch(`${tokenEndpoint}api/token`, {
+    method: 'POST',
+    body: JSON.stringify({
+        room_id: '633fcdd84208780bf665346a',
+        role: 'host',
+        user_id: '1234'
+    })
+});
+const { token } = await response.json();
+```
+
+### Disable 100ms token endpoint
+
+Due to some security concerns, if you don't wish to use the token endpoint to generate app tokens, then you can disable it on the [Developers page](https://dashboard.100ms.live/developer) on your dashboard by disabling the option "Disable &lt;room_id&gt;/&lt;role&gt; link format."
+
+![Token endpoint](/guides/disable-token-endpoint.png)

```

Comment:
this image can be zoomed in - impossible to tell what has been changed

Reply:
updated the image, check [here](https://100ms-docs-lh5x3ef6r-100mslive.vercel.app/docs/javascript/v2/foundation/security-and-tokens#disable-100ms-token-endpoint).